galley/pkg/config/source/kube/updater.go

```diff
@@ -0,0 +1,22 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kube
+
+import "istio.io/istio/galley/pkg/config/analysis/diag"
+
+// Updater updates resource statuses, based on the given diagnostic messages.
+type Updater interface {
```

Comment:
I wonder if this interface (and its method) should either be named more specific to the validation message case, since we want to eventually support other flavors of status field updates as well. That can probably wait, though.

Reply:
This is not needed anymore.